### PR TITLE
decode href in toc.yml/md to find file

### PR DESCRIPTION
**Pull request by superyyrrzz**
https://ceapex.visualstudio.com/Engineering/_workitems/edit/77465/

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs [hunk 1]
```diff
@@ -87,7 +87,8 @@ private void BuildCore(TocItemViewModel item, FileModel model, IHostService host
 
             void UpdateDependencies(HashSet<string> linkTos, Dictionary<string, ImmutableList<LinkSourceInfo>> linkSources, string link)
             {
-                var path = UriUtility.GetPath(link);
+                // var path = UriUtility.GetPath(link);
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: remove this line?

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: sure, forgot to remove this...

File: src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs [hunk 2]
```diff
@@ -87,7 +87,7 @@ private void BuildCore(TocItemViewModel item, FileModel model, IHostService host
 
             void UpdateDependencies(HashSet<string> linkTos, Dictionary<string, ImmutableList<LinkSourceInfo>> linkSources, string link)
             {
-                var path = UriUtility.GetPath(link);
+                var path = HttpUtility.UrlDecode(UriUtility.GetPath(link));
```
Comment (yufeih) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: see https://github.com/dotnet/docfx/pull/4246

`HtmlUtility.UrlDecode` turns `+` into ` `

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: Use `HttpUtility.UrlDecode` can keep consistent with all other parts in v2. Users can use `%2B` to work around this, or we can change it to `Uri.UnescapeDataString` throughout v2 later.

**Pull request merged**